### ISSUE TITLE
Q: What parameters would you recommend optimizing, and what could be the reason for some peaks not being called?

### ISSUE THREAD

**Issue by eregenyi** [taoliu/MACS#391]
Hi Taoliu,

I am using macs2 for CUT&RUN transcription factor peaks. My biologist colleagues pointed out some peaks that are not called, but are visible in the rpm-normalized bedgraph visualization, and I am trying to find an explanation, as well as find an optimal calling method. Some non-peaks may be called as well.

I used `--qvalue 0.05 --scale-to small --call-summits`.

As I understand the algorithm, a peak is not called if 1) the control has a lot of tags in the surrounding area (lambda 1k, 5k, 10k) or in total (lambda bg) 2) lack of bimodal character, so after shifting the Watson and Crick tags, they do not pile high at the binding site (in which case maybe it is not a real peak) 3) maybe there are duplicate tags in the peak, and it is shorter after removal of those.

I also computed the rpkm-ratio of the case vs control reads in each peak, and tried different cutoff values to filter a bit further.

Why do you think some peaks are not called? 
Which peak-calling parameters do you think I should adjust to find optimal values?
Do you think using a reasonable rpkm cutoff to filter out potential false positives is appropriate? 

Thank you for your kind answers in advance!

Best

**Comment by taoliu** [comment]
@eregenyi You can check if any of the three reasons are true.

1) the control has a lot of tags in the surrounding area (lambda 1k, 5k, 10k) or in total (lambda bg) 

Load both of the treat_pileup.bdg and control_lambda.bdg into a genome browser. You may need to convert them to bigWig files first. Check if the peaks that you colleagues pointed out indeed have large background noise.

2) lack of bimodal character, so after shifting the Watson and Crick tags, they do not pile high at the binding site (in which case maybe it is not a real peak) 

If you let MACS2 compute the tag shift (i.e. without `--nomodel` option),  you can check if the predicted fragment length is reasonable, and also if the figure from `_model.R` file looks good. It may cause trouble on peakcalling if the predicted fragment length is too short. Then the fragments can not be correctly piled up. 

In fact, the *model building* step is optional if you believe your data quality is good after certain quality control analyses. Usually, in order to uniformly process our data, we skip the *model building* and specify a fixed *fragment length* by `--nomodel --extsize 150`. 

3) maybe there are duplicate tags in the peak, and it is shorter after removal of those.

You can let MACS keep all duplicate reads by `--keep-dup all` and try to see the result. If the sequencing depth is ultra big, then the default setting in MACS2 to keep only one read per genomics position will kill the real enrichment. Alternatively, in this case, you can try down-sampling your data and run MACS2 again.

I don't have a recommendation for the reasonable rpkm cutoff to filter out false positives. Usually, we use FDR cutoff (like the q-value in MACS) and fold-change to filter out peaks.

**Comment by eregenyi** [comment]
Hi taoliu,

Thank you for your concise, detailed answers. 

Regarding the fragment lenghts, is that the same here as the lag between the watson and crick tags (denoted by _d_ in the publicaiton)? if so, then judging by the _model.R output, it is around 100 on average for this experiment. Because in CUT&RUN the MNase is "on leash", I think it could be shorter than in ChIP, but do you think it is a reasonable prediction?
Would you still specify  `--nomodel --extsize ...` if d was kind of d was kind of different (e.g. 20-30 bases apart) for the different samples?

Best

**Comment by taoliu** [comment]
@eregenyi Yes. That's the lag between the Watson and Crick tags. As I mentioned before, I would think the prediction of the fragment length as a step for quality control, instead of a necessity of actual peak calling. I don't know exactly what I should expect for CUT&RUN, but as long as it's not too short, such as around the read length -- as the "phantom peak" in [our ENCODE ChIP-seq guideline paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3431496/), your data should be fine. Although the 20-30 bases difference shouldn't affect the peak calling result too much, I still suggest `--nomodel --extsize ...` for all different samples in the same study, in order to make the processing consistent.

**Comment by ZHIDIHUAYUAN** [comment]
Hi,
I also have the same problem when calling peaks. I have ATAC-seq data, and I want to call peaks using macs2.
It seems that there is a peak at ~162500bp, but the result didn't call it as a peak，but I don't know why.
The reads were trimmed using trimmomatic, aligned using bowtie, and deduplicated using picard.
The command for calling peaks is : 
`macs2 callpeak -t SRR891268_hap1.sort.dedup.filt.sortbyname.fixed.tn5.final.bedpe -f BEDPE -n SRR891268_hap1 -g 4299059 --outdir macs2 -B --keep-dup all --call-summits`
I also had tried `-p 0.05` and `-p 0.01`, but it was also not called as a peak.
Please help me, thank you

**Comment by ZHIDIHUAYUAN** [comment]
Hi,
I also have the same problem when calling peaks. I have ATAC-seq data, and I want to call peaks using macs2.
It seems that there is a peak at ~162500bp, but the result didn't call it as a peak，but I don't know why.
The reads were trimmed using trimmomatic, aligned using bowtie, and deduplicated using picard.
The command for calling peaks is :
macs2 callpeak -t SRR891268_hap1.sort.dedup.filt.sortbyname.fixed.tn5.final.bedpe -f BEDPE -n SRR891268_hap1 -g 4299059 --outdir macs2 -B --keep-dup all --call-summits
I also had tried -p 0.05 and -p 0.01, but it was also not called as a peak.
Please help me, thank you
![image](https://user-images.githubusercontent.com/66869156/111744367-1577b180-88c6-11eb-8eff-df1ac8185efe.png)

**Comment by ZHIDIHUAYUAN** [comment]
Hi,
I found that when I set `--min-length` to 150, I got these peaks. I used the mode BEDPE, when I didn't set `--min-length`, it showed that the minimum length of peaks is assigned as the predicted fragment length "d", how BEDPE mode to predict the fragment length?

**Comment by ZHIDIHUAYUAN** [comment]
Hi,
I found that when I set --min-length to 150, I got these peaks. I used the mode BEDPE, when I didn't set  `--min-length`, it showed that the minimum length of peaks is assigned as the predicted fragment length "d". Does it mean that if the length of the peak is smaller than the fragment length of paired reads enriched in the peak, the peak will not be reported?
How should I choose the values of the `--min-length` ?

**Comment by ZHIDIHUAYUAN** [comment]
Hi,
I found that when I set --min-length to 150, I got these peaks. I used the mode BEDPE, when I didn't set `--min-length`, it showed that the minimum length of peaks is assigned as the predicted fragment length "d". Does it mean that if the length of the peak is smaller than the fragment length of paired reads enriched in the peak, the peak will not be reported?
How should I choose the values of the `--min-length` for ATAC-seq data ?

**Comment by taoliu** [comment]
Right. The `--min-length` sets the minimum length of peaks that will be called. It is used to avoid calling too short peaks. The choice can be arbitrary since in ATAC-seq we assume variable fragment sizes. Therefore, if you feel that the shorter peaks are still valid, try a smaller `--min-length` should be fine. You may also try to use a very small `--min-length` to get peaks with a wider range of different lengths and draw the distribution to help you decide.

**Comment by ZHIDIHUAYUAN** [comment]
Thank you for your apply.
But I still have some questions. For BEDPE mode, how does the software determine the min-length?
